ci/provisioning/post_provision_config_common_functions.sh

```diff
@@ -278,24 +297,6 @@ post_provision_config_nodes() {
         install_mofed
     fi
 
-    if [ -n "$INST_REPOS" ]; then
-        local repo
-        for repo in $INST_REPOS; do
-            branch="master"
-            build_number="lastSuccessfulBuild"
-            if [[ $repo = *@* ]]; then
-                branch="${repo#*@}"
-                repo="${repo%@*}"
-                if [[ $branch = *:* ]]; then
-                    build_number="${branch#*:}"
-                    branch="${branch%:*}"
-                fi
-            fi
-            local repo_url="${JENKINS_URL}"job/daos-stack/job/"${repo}"/job/"${branch//\//%252F}"/"${build_number}"/artifact/artifacts/$DISTRO_NAME/
-            dnf -y config-manager --add-repo="${repo_url}"
-            disable_gpg_check "$repo_url"
-        done
-    fi
     inst_rpms=()
```

Comment:
@brianjmurrell this has become a serious problem in the sense that currently all libfabric RPM CI runs do not use the correct RPM (the one that gets built within the PR), this patch fixes it. The PR repo has to be added before the dnf update.

Reply:
That can/should only be a problem if there is a libfabric somewhere that is newer than the one in the PR.  Is that the case here?  I would have thought our constantly moving to the very latest upstream release would have mitigated any such issue.

Do you have a run where you are seeing the wrong libfabric being installed that I can examine?

Reply:
Ahhh.  Or is the problem that `libfabric` is installed on the hosts even before we try to install DAOS and the DAOS install doesn't install the newest one, and the `dnf upgrade` doesn't upgrade it because it's done before the PR-repos are added?  Yes, that is probably the issue.

The other solution is to remove any daos dependencies here: https://github.com/daos-stack/daos/blob/a3c57e54ff6848ffc0ef384b60bbe27365ef3e78/ci/provisioning/post_provision_config_common_functions.sh#L302.

That said, I am not against adding the repos before the `dnf upgrade` in general.  I am just concerned about unintended consequences as we made a conscious decision to only want to upgrade the OS's packages, prior to any DAOS packages being installed at https://github.com/daos-stack/daos/blob/a3c57e54ff6848ffc0ef384b60bbe27365ef3e78/ci/provisioning/post_provision_config_common_functions.sh#L271.

I think before we could be sure such a change doesn't have any unintended consequences some testing of it needs to be done using the [build parameter](https://github.com/daos-stack/daos/blob/a3c57e54ff6848ffc0ef384b60bbe27365ef3e78/Jenkinsfile#L219-L221) and the `RPM-test-version:` commit pragma to request an older (than current) version of DAOS be installed for a test run and ensure that that version stays the installed one at the end of the provisioning.

Reply:
@brianjmurrell you can look at this run there: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-libfabric-PR-70-master/5/pipeline/194
this is the run from the libfabric PR repo and although the PR there was building libfabric-1.18.0-3 it ended up picking 1.18.0-2. Yes I think this only concerns libfabric currently as the RPM libfabric 1.15.0 is already installed on the system, then the dnf update runs and update to 1.18.0-2 then the PR repo gets added but it nevers gets updated again after that.